src/libraries/System.Private.CoreLib/src/System/Threading/Lock.cs

```diff
@@ -648,14 +647,22 @@ internal nint LockIdForEvents
 
         internal ulong OwningThreadId => _owningThreadId;
 
-        private static short DetermineMaxSpinCount() =>
-            AppContextConfigHelper.GetInt16Config(
-                "System.Threading.Lock.SpinCount",
-                "DOTNET_Lock_SpinCount",
-                DefaultMaxSpinCount,
-                allowNegative: false);
+        private static short DetermineMaxSpinCount()
+        {
+            if (IsSingleProcessor)
+            {
+                return 0;
```

Comment:
Is this the actual fix? - to force "no spin" in singleproc case, even if user specified something max spin count.

Reply:
Yea that's the actual fix